x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/action/TransportGetSearchApplicationAction.java

```diff
@@ -52,6 +54,16 @@ protected void doExecute(
                     for (String key : inconsistentIndices.keySet()) {
                         HeaderWarning.addWarning(key + " " + inconsistentIndices.get(key));
                     }
+                    if (searchApplication.hasStoredTemplate() == false) {
+                        HeaderWarning.addWarning(SearchApplication.NO_TEMPLATE_STORED_WARNING);
```

Comment:
Should the warning be added in the `SearchApplicationIndexService` so it is consistent with the other operations? We could even refactor this check into a method on that class...

Reply:
I think we only want to return this header on GET or PUT APIs.
The delete/list/search APIs do not need to return this header.
So it might make more sense to have this at the transport level.

Reply:
Right now I also have it on search and render metadata. I think this is OK, but WDYT?

Reply:
For now I'll update it, we can always add to render/search later.